### ISSUE TITLE
Simulate RGBD Camera

### ISSUE THREAD

**Issue by erickmu1** [UTRA-ART/Photon#5]
**Description:**
- Photon will need a variety of sensors to complete the competition, so we add them in simulation.

**Goals:**
- [x] Add (stereo?) camera plugin - check if there is an RGBD plugin
- [ ] Add `stereo_image_proc` to produce:
  - [ ] Depth maps
  - [x] Point Clouds
  - [x] Rectified Images

**Acceptance Criteria:**
- [x] Relevant camera topics (e.g. images) are published
- [x] Camera outputs change when there is a change in the environment

**Notes:**
- It would be nice to keep track of what plugin arguments can be modified so we can more accurately simulate our sensors in the future.
- Depending on the camera we decide on for Photon, we may need to emulate some extra functionality.

**Comment by erickmu1** [comment]
To test whether the stereo camera is created properly, you need to be able to reproduce the following in RViz (note how the robot model in RViz aligns *perfectly* with the parts of the robot seen in the image)
![image](https://user-images.githubusercontent.com/40526591/115943785-f9949c00-a47f-11eb-940e-1b7b8859beaa.png)

**Comment by erickmu1** [comment]
This may be good to put somewhere else but at least for reference, PR #15 will have the code that does this (using the Gazebo stereo camera plugin) correctly. The idea is as follows:

- Naturally, one would make a `camera_link` and connect the plugin using that frame. This is **not** what you actually want to do.
- Rather, the [`multicamera`](http://gazebosim.org/tutorials?tut=ros_gzplugins#Multicamera) (stereo) Gazebo plugin assumes that the frame given is at the location of the left camera.
- Due to Computer Vision conventions adopted in ROS, you also need an optical frame for your left camera (http://www.ros.org/reps/rep-0103.html#suffix-frames)

In short, you will need to define the following links in your URDF:
```
camera_link  # Used for placing your camera on your robot
camera_left_link  # The actual location of the left camera within your stereo camera
camera_left_link_optical  # The optical link following ROS conventions at the left camera
```

The parameter that specifies the distance between the left and right cameras (of the stereo camera) is know as the `hackBaseline`. Note that generally this parameter is smaller than how big the stereo camera is (since the left/right cameras must be inside the stereo camera). I rather prefer to refer to the "hack baseline" as the "stereo offset".

**Comment by erickmu1** [comment]
`stereo_im_proc` publishes disparity images. If we decide to use RTABmap again, we can use it's nodelets to create depth images and do visual odom. Hence creating (and visualizing) depth images is deferred until then)
